### PR TITLE
[photon-targeting][photon-lib] Add tests to the targeting classes and cleanup photon-lib & photon-targeting

### LINKED ISSUE #1004: Move C++ targetting classes into photon-targetting
Currently, PhotonTargetting classes live in photon-lib in C++ while they live in photon-targetting in Java. This should be updated so that the C++ is moved into photon-targetting.

### PR DESCRIPTION
**Pull request by srimanachanta**
prep and bring some work from #974 (hence why I'm not adding Packet tests)

Also makes the currently plural targeting classes singular as they should be

### CODE REVIEW

File: photon-lib/src/main/java/org/photonvision/PhotonCamera.java [hunk 1]
```diff
@@ -111,20 +105,14 @@ public void close() {
...
     public static void setVersionCheckEnabled(boolean enabled) {
         VERSION_CHECK_ENABLED = enabled;
     }
 
     Packet packet = new Packet(1);
 
-    private static AprilTagFieldLayout lastSetTagLayout =
-            AprilTagFields.kDefaultField.loadAprilTagLayoutField();
-
-    // Existing is enough to make this multisubscriber do its thing
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/PhotonCamera.java:125]: Why are we removing this

File: photon-lib/src/main/java/org/photonvision/simulation/VideoSimUtil.java [hunk 2]
```diff
@@ -374,8 +374,8 @@ public static void drawTagDetection(int id, Point[] dstPoints, Mat destination)
     /**
      * Set the field dimensions that are used for drawing the field wireframe.
      *
-     * @param fieldLengthMeters
-     * @param fieldWidthMeters
+     * @param fieldLengthMeters field length in meters
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/simulation/VideoSimUtil.java:377]: This comment is not very helpful. Can we maybe also say how width/length correspond to X and Y?

File: photon-lib/src/main/java/org/photonvision/PhotonCamera.java (hunk 1 above)
Comment (srimanachanta) [reply photon-lib/src/main/java/org/photonvision/PhotonCamera.java:125]: Currently unused and I will add back all this logic when we do protobuf ATFL

Comment (mcm001) [reply photon-lib/src/main/java/org/photonvision/PhotonCamera.java:125]: Like I said in my comment, this field is _not_ unused.

Comment (srimanachanta) [reply photon-lib/src/main/java/org/photonvision/PhotonCamera.java:125]: Oh sorry, I thought you were refering to the ATFL field, yes I accidentally removed the multisubscriber

**Review by mcm001 — APPROVED** [review]

**Pull request merged**